src/main/java/com/aws/greengrass/shadowmanager/sync/RequestUnboundedQueue.java

```diff
@@ -16,23 +16,14 @@
 import javax.inject.Inject;
 
 /**
- * Blocking "queue" implementation that keeps a single request per shadow. If a request comes in for a shadow, it is
- * merged together with the existing request.
- * <p/>
- * This class is thread safe and implements the useful methods of the {@link java.util.concurrent.BlockingQueue}
- * interface. However, this class intentionally does not implement the interface as it is not a Collection. It is not
- * iterable as the iterator needs to be synchronized to avoid concurrent modifications. This is not possible with the
- * current implementation (but storing keys in a separate linked list could be used to solve this).
+ * Unbounded "queue" implementation that keeps a single request per shadow. If a request comes in for a shadow, it is
+ * merged together with the existing request. This queue will expand in size to accommodate all shadows sent to it.
  * <p/>
  * For the simplest blocking use case, threads can call {@link #take()} and they will wait until items are available.
- * Threads can call {@link #put(SyncRequest)} and they will wait until space is available to insert.
+ * There are no blocking cases when adding to the queue.
  */
-public class RequestBlockingQueue {
+public class RequestUnboundedQueue {
 
-    /**
-     * Default maximum number of queued requests.
-     */
-    static final int MAX_CAPACITY = 1024;
```

Comment:
I understand that we will never reach this situation, but how are we handling the case where java throws an out of memory error when shadow manager tries to put a new request to the requests map?

Reply:
Never say never. You're right in the fact that we could hit this case. I'll take a look at the code tomorrow and try to add in a try-catch (or multiple) to account for this. The most sensible place I can see is on the `put` call as the map should resize when inserting an element causes it to go above load factor.